tests/kokkos-based/helpers.hpp

```diff
@@ -108,8 +108,37 @@ mdspan_t make_mdspan(ValueType *data, std::size_t ext0, std::size_t ext1) {
   return mdspan_t(data, ext0, ext1);
 }
 
+template <typename RealValue>
+KOKKOS_INLINE_FUNCTION
+RealValue scalar_abs_diff(RealValue v1, RealValue v2) {
+  return std::abs(v2 - v1);
+}
+
+template <typename T>
+T scalar_abs_diff(const std::complex<T> &v1, const std::complex<T> &v2) {
+  const auto dr = scalar_abs_diff(v1.real(), v2.real());
+  const auto di = scalar_abs_diff(v1.imag(), v2.imag());
+  return std::max(dr, di);
```

Comment:
@dalg24 This is the error measure that LAPACK likes to use for complex numbers.  I think the original motivation was to avoid square roots.

Reply:
no it would be sqrt(8)
https://godbolt.org/z/PoqPobhWT

Reply:
@dalg24 Sorry, somehow I didn't catch that you mean the whole routine...

@dalg24 @mhoemmen 
While `sqrt` stays high on my list, my issue was that `Kokkos::abs()` has no overload for `std::complex` and `std::abs()` has no overload for `Kokkos::complex`. You made me realize we wouldn't need `scalar_diff_abs()` if we had `abs()` that handles both complex types (we extended `conj_if_needed()` this way). Should we introduce that ? If so, should it be in a separate header so the Kokkos algorithm implementations can also use it ?

Reply:
Ok, we don't need extra `abs()` for this: please see https://github.com/kokkos/stdBLAS/pull/193/commits/80a8e2cf24bab00be602c99c6c1cc43259d01f2d. Thank you @dalg24!

Reply:
@MikolajZuzek It's OK to work around the missing Kokkos feature for now, but any changes to `Kokkos::abs` should happen in Kokkos proper, not here.  Otherwise, you'll get build errors here if `Kokkos::abs` ever adds an overload for `std::complex`.

In brief: Please do not add overloads to Kokkos functions (like `Kokkos::abs`) here.  The Kokkos repository is the right place to add them.

Reply:
Thanks @mhoemmen - totally agree. Fortunately, on https://github.com/kokkos/stdBLAS/commit/80a8e2cf24bab00be602c99c6c1cc43259d01f2d, I was able to just use `std::abs()` and `Kokkos::abs()` instead of introducing new thing for scalar diff - which is what, I think, @dalg24 originally suggested.